### ISSUE TITLE
docker-compose -H does not work with compose 2.11.0

### ISSUE THREAD

**Issue by niklas-rittmann** [docker/compose#9852]
---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------

**Description**

The remote socket support stopped working with compose 2.11.0. 

**Steps to reproduce the issue:**
1. Create an example docker-compose file
2. Try to start the container at a remote host using `docker-compose -H ssh://<<Remote host to deploy at>> up`

**Describe the results you received:**
Instead of deploying to the remote server it deploys on the localhost.

**Describe the results you expected:**
Deploy the container on the remote host.

**Additional information you deem important (e.g. issue happens only occasionally):**

Using `DOCKER_HOST=ssh://<<Remote host to deploy at>> docker-compose up` still works as expected.

**Output of `docker compose version`:**

```
Docker Compose version v2.11.0
```

**Output of `docker info`:**

```
Client:                                                                                                                                                                                         
 Context:    default                                                                                                                                                                            
 Debug Mode: false                                                                                                                                                                              
 Plugins:                                                                                                                                                                                       
  buildx: Docker Buildx (Docker Inc., v0.9.1)                                                                                                                                                   
  compose: Docker Compose (Docker Inc., v2.11.0)                                                                                                                                                
                                                                                                                                                                                                
Server:                                                                                                                                                                                                                                                                                                                                                                    
 Server Version: 20.10.18                                                                                                                                                                       
 Storage Driver: overlay2                                                                                                                                                                          
```

**Additional environment details:**

**Comment by laurazard** [comment]
@ulyssessouza looks like this was introduced in https://github.com/docker/compose/pull/9817. Can you take a look?

**Comment by sphinxc0re** [comment]
I'm unsure if this is related, but I'm having similar issues with the `--context` option:

This lists my remote containers:
```
docker --context my_context ps
```

This does not:
```
docker --context my_context compose -p my_fixed_project_name ps
```

**Comment by niklas-rittmann** [comment]
Don’t know if this helps but:
`docker -H <<Remote host>> compose up` also does not work.

**Comment by sphinxc0re** [comment]
This makes compose unsuable for me.

**Comment by laurazard** [comment]
@sphinxc0re in the meantime, as mentioned by @niklas-rittmann, you can use `DOCKER_HOST=ssh://<<Remote host to deploy at>> docker compose [xxx]` to circumvent this issue.

**Comment by sphinxc0re** [comment]
Nope, this does not work either.

**Comment by sphinxc0re** [comment]
Just tested this again with `-p my_project_name` specified. The workaround works now.

**Comment by kaangoksal** [comment]
Also doesn't work for me 

```
(venv) kaan@ubuntu-vm:~/Desktop/DjangoProject$ DOCKER_HOST=ssh://ubuntu@10.0.0.57 docker compose  -f docker-compose.base.yml -f deploy/docker-compose.yml --env-file deploy/env_file ps

Cannot connect to the Docker daemon at http://docker.example.com. Is the docker daemon running?
```

**Comment by milas** [comment]
This is fixed by #9871, which will be in the next Compose release (which I'm in the process of tagging 🙂).
